src/DittoMachine.sol

```diff
@@ -316,16 +321,24 @@ contract DittoMachine is ERC721, ERC721TokenReceiver {
      * @dev only use it for a minted clone.
      */
     function _getMinAmount(CloneShape memory cloneShape) internal view returns (uint256) {
+        uint256 floorId = uint256(keccak256(abi.encodePacked(
```

Comment:
What do you think about this?

We pass in floor clone's worth as a parameter here. This will save gas in computing the `floorId` which we already have in `duplicate()` when we call `_getMinAmount()`.

We also call this function in `getMinAmountForCloneTransfer()`. We can compute floor clone's worth there to pass as a parameter.

Reply:
can be done in another PR.

Reply:
In `duplicate` I moved the other `floorId` calculation to the first branch of the if statement, so with `_getMinAmount` in the second branch of the if statement `floorId` is only calculated once for each branch..